enhancements/network/persistent-ips-sdn-secondary-networks.md

```diff
@@ -0,0 +1,1012 @@
+  - "@trozet"
+  - "@dougbtv"
+  - "@jcaamano"
+  - "@AlonaKaplan"
+approvers: # A single approver is preferred, the role of the approver is to raise important questions, help ensure the enhancement receives reviews from all applicable areas/SMEs, and determine when consensus is achieved such that the EP can move forward to implementation.  Having multiple approvers makes it difficult to determine who is responsible for the actual approval.
+  - "@trozet"
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - "@dougbtv"
+  - "@AlonaKaplan"
+creation-date: 2023-08-16
+last-updated: 2023-08-16
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/SDN-4035
+see-also:
+  - "/enhancements/this-other-neat-thing.md"
+replaces:
+  - "/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# Persistent IPs for SDN secondary networks
+
+## Summary
+
+This enhancement proposal focuses on tying IP address allocation to Virtual
+Machines instead of the pods where they run. In short, the IP allocation
+should be available during the Virtual Machine lifecycle, rather than the
+lifecycle of the pod where it runs on.
+
+## Motivation
+
+Openshift Virt customers want to offload IPAM from their custom solutions
+(e.g. custom DHCP server running on their cluster network) to SDN.
+
+The other alternative - static IP address allocation - is also not ideal for
+virtualization users, given how cumbersome it is - i.e. specifying the IP for
+each VM's interface.
+
+### User Stories
+
+- As an Openshift Virtualization user, I want my VMs to automatically get IP
+addresses from a configured range, so I don't need to manage my own IP pools,
+DHCP server, or static IP configuration.
+- As an Openshift Virtualization user, I want to migrate (one of) my VMs to
+another node, preserving the networking configuration, so the IP-aware
+applications running in the VM are not impacted.
+- As an Openshift Virtualization user, I want to start Virtual Machines
+(currently stopped) with the original VM network configuration, so the IP-aware
+applications running in the VM are not impacted.
+- As an Openshift network administrator, I want to decide which networks can be
+subject to having persistent IP allocations surviving VM migration, and VM stop.
+
+### Goals
+
+- Manage Openshift Virtual Machines IP addresses from within the SDN solution
+provided by OVN-Kubernetes.
+- IPv4 range can be defined on OVN Kubernetes secondary network
+- VMs are able to get a IP from this subnet
+- The IP sticks to a VM during live-migration
+- The IP sticks to a VM even if it is temporarily shut down
+
+### Non-Goals
+
+- IPv6 support is not required.
+- This enhancement's scope is limited to secondary networks - i.e. the default
```

Comment:
Just a quick note that I did not see this clarified in the lastest update, in case it was forgotten. Also ...

I guess that the `IPAMClaim` specification does not make any distinction on network types or topologies, so it would be us downstream explicitly limiting that support only for l2 / localnet topologies.

Meaning if we were to fully support the `IPAMClaim` specification we should do so for all networks, shouldn't we?

Reply:
I'd limit ourselves to the ask we got from customers and only implement it for the 2 configurations we know people have expressed interest on.